File: 202-happy-number/happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        s = set()

        while n not in s:
            s.add(n)
            sum = 0
            
            while n != 0:
                dig = n % 10
                sum = sum + dig ** 2
                n = n // 10
            n = sum
        return n == 1

File: 202-happy-number/happy-number.cpp
class Solution {
public:
    bool isHappy(int n) {
        set<int> s;

        while(s.find(n) == s.end()) {
            int sum = 0;

            s.insert(n);
            while(n != 0) {
                int dig = n % 10;
                n = n / 10;
                dig = pow(dig, 2);
                sum = sum + dig;
            }
            n = sum;
        }
        if (n == 1) {
            return true;
        } else {
            return false;
        }
    }
};